### PR TITLE
#207 Added build schemes

### LINKED ISSUE #207: Add build schemes
## Goal

To avoid future issues of not submitting proper production-ready code, we should create two build schemes in Xcode, one for development purposes (e.g. local builds, TestFlight) and one for production code (e.g. App Store, TestFlight as well). Suggested by @AAAstorga

### Walkthrough Guide
[How to set up multiple schemes & configurations in Xcode for your React Native iOS app](https://zeemee.engineering/how-to-set-up-multiple-schemes-configurations-in-xcode-for-your-react-native-ios-app-7da4b5237966)

### Helpful Stack Overflow Article
[Swift: how to use PREPROCESSOR Flags (like `#if DEBUG`) to implement API keys?](https://stackoverflow.com/questions/38813906/swift-how-to-use-preprocessor-flags-like-if-debug-to-implement-api-keys)

### PR DESCRIPTION
**Pull request by mattbarker016**
I added two new schemes within the Xcode project. I would love if we could add a build script that shows a pop-up to remind developers to check whether they're using the proper scheme with archiving. And if this could include a reminder and syncing / incrementing build numbers, bless up.

### Production
Uses main production server for Network requests. This scheme should be used for all App Store uploads.

### Staging
Uses development server for Network requests. This will also archive and profile with the development server, so we can beta-test changes made on the development server through TestFlight. 

## Relevant Code in `Endpoints.swift`
```

    // MARK: Global Network Variables

    static let apiVersion = "v1"
    
    static var ipAddress: String {
        // For local testing, use "http://\(localIPAddress):3000/api/\(apiVersion)/"
        guard let baseURL = Bundle.main.object(forInfoDictionaryKey: "SERVER_URL") as? String else {
            fatalError("Could not find SERVER_URL in Info.plist!")
        }
        return baseURL
    }

    /// Network address being used within app, defined by schemes and build configurations.
    static var address: String {
        print("[Network] Using", ipAddress)
        return "https://\(ipAddress)/api/\(apiVersion)"
    }
```

Resolves #207.

### CODE REVIEW

**Comment by mattbarker016** [comment]
@ywsang @Omarrasheed @AAAstorga bump on this PR reviews. Would like to get this merged to Yana can pull for hers!

**Comment by mattbarker016** [comment]
Also, can someone please clone this branch and see if the schemes appear? They don't on my new computer

File: TCAT/Supporting Files/Info.plist [hunk 1]
```diff
@@ -2,6 +2,8 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
+	<key>SERVER_URL</key>
```
Comment (ywsang) [inline TCAT/Supporting Files/Info.plist:5]: shouldn't we have two keys, `LOCAL_SERVER_URL` and `RELEASE_SERVER_URL`

File: TCAT/Utilities/Endpoints.swift [hunk 2]
```diff
@@ -18,54 +18,30 @@ enum NetworkType: String {
...
-        if isTestFlight() {
-           return debugIPAddress
-        }
-        #endif
-        switch networkType {
-        case .local: return localIPAddress
-        case .debug: return debugIPAddress
-        case .release: return releaseIPAddress
+        // For local testing, use "http://\(localIPAddress):3000/api/\(apiVersion)/"
+        guard let baseURL = Bundle.main.object(forInfoDictionaryKey: "SERVER_URL") as? String else {
```
Comment (ywsang) [inline TCAT/Utilities/Endpoints.swift:35]: if we have the two keys, here we can do
```
if #DEBUG 
let key = "LOCAL_SERVER_URL"
#else 
let key = "SERVER_URL"
#endif
guard let baseURL = Bundle.main.object(forInfoDictionaryKey: key) as? String else {...}
```

**Review by ywsang — COMMENTED** [review]
have two questions

**Comment by ywsang** [comment]
also, I don't see the schemes 
Screen Shot 2019-03-21 at 10 34 12 PM

File: TCAT/Utilities/Endpoints.swift (hunk 2 above)
Comment (mattbarker016) [reply TCAT/Utilities/Endpoints.swift:35]: image

`SERVER_URL` changes based on the scheme! This is on the bottom of the Build Settings page.

**Comment by mattbarker016** [comment]
@ywsang Give it a shot now!

**Review by kevinchan159 — APPROVED** [review]
lgtm

**Review by Omarrasheed — APPROVED** [review]
LGTM

**Pull request merged**